android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIText.java

```diff
@@ -135,10 +137,16 @@ public void onLayoutChange(View v, int left, int top, int right, int bottom, int
 		} else {
 			tv.setGravity(Gravity.TOP | Gravity.LEFT);
 		}
+		tv.setBackgroundColor(Color.TRANSPARENT);
 
 		textInputLayout = new TextInputLayout(proxy.getActivity());
-		textInputLayout.addView(tv, new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.WRAP_CONTENT));
-		setNativeView(textInputLayout);
+		textInputLayout.addView(tv, new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.MATCH_PARENT));
+
+		nestedScrollView = new NestedScrollView(proxy.getActivity());
+		nestedScrollView.setFillViewport(true);
+		nestedScrollView.addView(textInputLayout);
+
+		setNativeView(nestedScrollView);
```

Comment:
We'll need to double check if Titanium's `setTouchEnabled(false)` call will gray out the TextField/TextArea like it did before TextInputLayout was implemented. (I forgot to check this before.)

Reply:
I just verified that setting `touchEnabled` property to `false` no longer disable TextFields/TextAreas. In fact, they can still be edited.

The best solution to this may be to duplicate what Google has done in their `TextInputLayout.java` code where they override the setEnabled() method and pass that state to its children via our own NestedScrollView derived class.
https://github.com/android/platform_frameworks_support/blob/master/design/src/android/support/design/widget/TextInputLayout.java#L843

However, there is another issue. Animated hint text (the text above the TextField/TextArea) can be scrolled off the view because we're wrapping the TextInputLayout within a NestedScrollView. The solution may be to dump the NestedScrollView (which fixes the above `enabled` issue) and have the `TiUIEditText` class implement the `NestedScrollingChild` interface.